packages/vite/src/devFeServer.ts

```diff
@@ -80,6 +80,30 @@ async function createServer() {
       : route.pathDefinition
 
     app.get(expressPathDef, createServerAdapter(routeHandler))
+
+    app.all(
+      '/_rw_mw',
+      createServerAdapter(async (req: Request) => {
+        const entryServerImport = await vite.ssrLoadModule(
+          rwPaths.web.entryServer as string // already validated in dev server
+        )
+
+        const middleware = entryServerImport.middleware
+
+        let out = null
+        if (middleware) {
+          try {
+            out = await middleware(req)
+          } catch (e) {
+            console.error('Whooopsie, error in middleware POST handler')
+            console.error(e)
+          }
+        }
+
+        // @TODO: We should check the type of resposne here I guess
```

Comment:
Future danny seemed like a better fit for this 🤣. I'll generalise this into a function that invokes, validates and responds

Reply:
Actually - question on this, what do you think of Remix's pattern of "throwing" Redirects? 

```js
try {
mwRes = invokeMiddleware(middleware, req) || mwRes.next()
} catch(e) {
	if (e.type === "redirect")  {
		return e.toResponse()
  } else {
   console.log('Error invoking middleware > \n', e)
  mwRes = mwRes.next()
  }
}
```

This is considered an anti pattern in some circles (pointed out by Rob) because we're using throwing errors to control flow

Reply:
Yeah, I'm one of those "it's an anti-pattern" ppl. But React is also doing it for their Suspense stuff to work, right? So maybe it's not an anti pattern in the JS world. But to me throwing errors should really only be done under exceptional circumstances to signal that something is really wrong

Reply:
Hmmm, I'll keep it as is for now then!  

Updated the implementation just now with the new invoke helper function